cli/src/main/java/com/box/l10n/mojito/cli/command/PushCommand.java

```diff
@@ -87,6 +96,16 @@ public void execute() throws CommandException {
         List<PollableTask> pollableTasks = new ArrayList<>();
 
         ArrayList<FileMatch> sourceFileMatches = commandHelper.getSourceFileMatches(commandDirectories, fileType, sourceLocale, sourcePathFilterRegex);
+
+        // filter out source files which do not contain diff changes if git diff is enabled.
+        if (gitDiff) {
+            sourceFileMatches = getGitChangeFiles(sourceFileMatches);
+            if (sourceFileMatches.isEmpty()) {
+                logger.debug("No new string has been created. No need to create branch and process asset.");
+                return;
```

Comment:
i'd rather keep function with a single return

Reply:
remove line 103 - 105 to keep single return